### PR TITLE
Move the URL search param processing to sveltekit-search-params  and to the top-level +page file for Logs

### PR DESCRIPTION
**Pull request by matthampton-stock2shop**
This change separates the active filters (filterTerms) from the constant list of available filters (facets), and the returned aggregations.

This turns the filterTerms in to a simple record type that is easy to bind directly to the store that is created by sveltekit-search-params queryParameters function.

I set that upon the +page component which feels right.

### CODE REVIEW

**Comment by TomRadford** [comment]
@matthampton-stock2shop shall I close #491 ?

**Review by TomRadford — APPROVED** [review]
Happy to merge

**Review by flumonion — COMMENTED** [review]
This looks good to me, well done.
The only thing I am not a fan of is the word `AggregationStats` in the model.

It is the stats but most importantly the buckets.
Is this not the `AggregationResponse`?

Stats makes me think buckets are not included. Also, stats is an ES keyword which [means something else](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-stats.html)...

File: src/components/ES/Aggregations.svelte [hunk 1]
```diff
@@ -5,12 +5,12 @@
   import Btn from '@/components/form/Btn.svelte';
   import Aggregation from '@/components/ES/Aggregation.svelte';
   import type { Agg } from '@/models/es/aggregations';
-  import { createEventDispatcher } from 'svelte';
-
-  const dispatch = createEventDispatcher();
+  import type { AggregationsResponse, FilterTerms } from '@/models/es/data';
 
   export let isLoading: boolean;
   export let aggs: Agg[];
```
Comment (matthampton-stock2shop) [inline src/components/ES/Aggregations.svelte:11]: rename aggs -> facets
rename Agg -> Facet

File: src/components/ES/Aggregations.svelte [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
+  export let aggsResponse: AggregationsResponse;
```
[the remark this answers is not in the document]
Comment (TomRadford) [reply src/components/ES/Aggregations.svelte]: 28c89575bcba2fba2899a40c54742bc5843e37f

File: src/components/ES/Aggregations.svelte [hunk 3]
```diff
@@ -45,12 +46,12 @@
...
 </div>
 
-{#each aggs as agg}
+{#each facets as facet}
   <div class="border-t border-neutral-200 p-4">
-    <Aggregation
-      {agg}
-      bind:value={filterTerms[agg.key]}
-      aggResponse={aggsResponse[agg.key]}
+    <AggregationComponent
```
Comment (matthampton-stock2shop) [inline src/components/ES/Aggregations.svelte:51]: Hmm.. I'm creating more work here... but the truth is that this is a 'FacetComponent' (I think).... Easy to rename?

Can of worms?

File: src/components/Logs/Logs.svelte [hunk 4]
```diff
@@ -74,7 +76,7 @@ Manages the layout of the logs page.
 <div id="page" class="flex h-full justify-between">
   <!--  Sidebar-->
   <div class="mr-6 h-full min-w-[260px] max-w-[260px] overflow-y-auto rounded bg-white shadow">
-    <Aggregations {aggs} bind:filterTerms aggsResponse={data?.aggregations || {}} {isLoading} />
+    <Aggregations {facets} bind:filterTerms aggsResponse={data?.aggregations || {}} {isLoading} />
```
Comment (matthampton-stock2shop) [inline src/components/Logs/Logs.svelte:79]: And I suppose this is actually a 'FacetsComponent' ...

File: src/lib/es/types.ts [hunk 5]
```diff
@@ -181,11 +181,11 @@ export type Order = 'asc' | 'desc';
 /**
  * Elastic search meta data
  */
-export interface ESData {
+export interface ESData<H = Hits> {
```
Comment (matthampton-stock2shop) [inline src/lib/es/types.ts:184]: looks nice

File: src/lib/es/types.ts [hunk 6]
```diff
@@ -181,11 +181,11 @@ export type Order = 'asc' | 'desc';
...
   took?: number;
   timed_out?: boolean;
   _shards?: Shards;
-  hits?: Hits;
+  hits?: H;
   aggregations?: { [key: string]: Aggregation };
```
Comment (matthampton-stock2shop) [inline src/lib/es/types.ts:189]: `aggregations?: Aggregations` ?

File: src/components/ES/Aggregations.svelte (hunk 3 above)
Comment (TomRadford) [reply src/components/ES/Aggregations.svelte]: d8502dd2c0404d887d22ac9bd14aaa5bd415a717

File: src/components/Logs/Logs.svelte (hunk 4 above)
Comment (TomRadford) [reply src/components/Logs/Logs.svelte]: d8502dd2c0404d887d22ac9bd14aaa5bd415a717

File: src/lib/es/types.ts (hunk 6 above)
Comment (TomRadford) [reply src/lib/es/types.ts]: 739a885bf267fbaa90329302baf649e2776c910e

**Review by TomRadford — APPROVED** [review]

**Pull request merged**